com.sap.cloud.lm.sl.cf.database/src/test/java/com/sap/cloud/lm/sl/cf/database/migration/executor/type/DatabaseTypeSetterFactoryTest.java

```diff
@@ -0,0 +1,90 @@
+    @Test
+    public void testGetWithCustomRegisteredTypeSettersAndSingleMatch() {
+        List<DatabaseTypeSetter> registeredDatabaseTypeSetters = Arrays.asList(new BooleanDatabaseTypeSetter());
+        databaseTypeSetterFactory = new DatabaseTypeSetterFactory(registeredDatabaseTypeSetters);
+
+        DatabaseTypeSetter resultDatabaseTypeSetter = databaseTypeSetterFactory.get(BOOL_DEFAULT_DATABASE_TYPE_SETTER);
+
+        Assertions.assertEquals(registeredDatabaseTypeSetters.get(0), resultDatabaseTypeSetter);
+    }
+
+    @Test
+    public void testGetWithCustomRegisteredTypeSettersWhenNoMatchingTypeSetter() {
+        List<DatabaseTypeSetter> registeredDatabaseTypeSetters = Arrays.asList(new BooleanDatabaseTypeSetter());
+        databaseTypeSetterFactory = new DatabaseTypeSetterFactory(registeredDatabaseTypeSetters);
+
+        Assertions.assertThrows(IllegalStateException.class, () -> databaseTypeSetterFactory.get(NON_EXISTING_DATABASE_TYPE_SETTER));
+    }
+
+    @Test
+    public void testGetWithCustomRegisteredTypeSettersWhenMultipleMatchingTypeSetters() {
+        List<DatabaseTypeSetter> registeredDatabaseTypeSetters = Arrays.asList(new BooleanDatabaseTypeSetter(),
+                                                                               new BooleanDatabaseTypeSetter());
+        databaseTypeSetterFactory = new DatabaseTypeSetterFactory(registeredDatabaseTypeSetters);
+
+        Assertions.assertThrows(IllegalStateException.class, () -> databaseTypeSetterFactory.get(BOOL_DEFAULT_DATABASE_TYPE_SETTER));
+    }
+
+    @Test
+    public void testGetWithDefaultRegisteredTypeSettersWhenMatchingBoolTypeSetter() {
+        databaseTypeSetterFactory = new DatabaseTypeSetterFactory();
+
+        DatabaseTypeSetter resultDatabaseTypeSetter = databaseTypeSetterFactory.get(BOOL_DEFAULT_DATABASE_TYPE_SETTER);
+
+        Assertions.assertTrue(resultDatabaseTypeSetter.getSupportedTypes()
+                                                      .contains(BOOL_DEFAULT_DATABASE_TYPE_SETTER));
+    }
+
+    @Test
+    public void testGetWithDefaultRegisteredTypeSettersWhenMatchingLongTypeSetter() {
+        databaseTypeSetterFactory = new DatabaseTypeSetterFactory();
+
+        DatabaseTypeSetter resultDatabaseTypeSetter = databaseTypeSetterFactory.get(LONG_DEFAULT_DATABASE_TYPE_SETTER);
+
+        Assertions.assertTrue(resultDatabaseTypeSetter.getSupportedTypes()
+                                                      .contains(LONG_DEFAULT_DATABASE_TYPE_SETTER));
+    }
+
+    @Test
+    public void testGetWithDefaultRegisteredTypeSettersWhenMatchingStringTypeSetter1() {
+        databaseTypeSetterFactory = new DatabaseTypeSetterFactory();
+
+        DatabaseTypeSetter resultDatabaseTypeSetter = databaseTypeSetterFactory.get(STRING_DEFAULT_DATABASE_TYPE_SETTER_1);
+
+        Assertions.assertTrue(resultDatabaseTypeSetter.getSupportedTypes()
+                                                      .contains(STRING_DEFAULT_DATABASE_TYPE_SETTER_2));
```

Comment:
This assertion actually tests the type setter itself (whether it supports both string types) and not the factory. Change it to `Assertions.assertTrue(resultDatabaseTypeSetter.getSupportedTypes().contains(STRING_TYPE));` and remove the `STRING_DEFAULT_DATABASE_TYPE_SETTER_2` constant.